From a028e4e8739d994f25b288c0798a543db68199fd Mon Sep 17 00:00:00 2001
From: Giuseppe Lavagetto <glavagetto@wikimedia.org>
Date: Tue, 24 Sep 2019 07:58:26 +0200
Subject: [PATCH] Translate the default section name to DEFAULT in
 ReadOnlyBySection too

Bug: T233679
Index: b/conftool/extensions/dbconfig/config.py
===================================================================
--- a/conftool/extensions/dbconfig/config.py	2019-09-24 11:50:11.778266782 +0200
+++ b/conftool/extensions/dbconfig/config.py	2019-09-24 11:50:11.778266782 +0200
@@ -94,7 +94,8 @@
         for obj in sections:
             if not obj.readonly:
                 continue
-            config[obj.tags['datacenter']]['readOnlyBySection'][obj.name] = obj.ro_reason
+            section = self._mw_section(obj.name)
+            config[obj.tags['datacenter']]['readOnlyBySection'][section] = obj.ro_reason
 
         # now fill them with the content of instances
         for instance in instances:
@@ -111,12 +112,7 @@
                     continue
 
                 fraction = section['percentage']/100
-                # Mangle the key.
-                # Thanks for this, MediaWiki
-                if section_name == self.default_section:
-                    section_key = 'DEFAULT'
-                else:
-                    section_key = section_name
+                section_key = self._mw_section(section_name)
 
                 main_weight = int(section['weight'] * fraction)
                 section_load_index = 1
@@ -452,3 +448,12 @@
                 name=name, masters=sorted(section[0].keys())))
 
         return errors
+
+    def _mw_section(self, section):
+        """Translates the section name in the one expected by MediaWiki."""
+        # Mangle the section key.
+        # Thanks for this, MediaWiki
+        if section == self.default_section:
+            return 'DEFAULT'
+        else:
+            return section
diff --git a/conftool/tests/unit/test_dbconfig.py b/conftool/tests/unit/test_dbconfig.py
index 010156e..53a942a 100644
--- a/conftool/tests/unit/test_dbconfig.py
+++ b/conftool/tests/unit/test_dbconfig.py
@@ -429,7 +429,7 @@ class TestDbConfig(TestCase):
                 'DEFAULT': [{'db3': 10}, {'db1': 10, 'db2': 10}],
                 's4': [{'db2': 10}, {}]},
              'groupLoadsBySection': {},
-             'readOnlyBySection': {'s3': 'Some reason.'}}
+             'readOnlyBySection': {'DEFAULT': 'Some reason.'}}
         }
         res1 = self.config.compute_config(sections, instances)
         self.assertEqual(res1, expected)
